src/modules/heroku.rs

```diff
@@ -0,0 +1,501 @@
+            let mut context = default_context();
+            context.env.insert("HEROKU_HOST", "will.be.ignored".into());
+            context
+                .env
+                .insert("HEROKU_GIT_HOST", "other.git.host".into());
+
+            assert_eq!(
+                git_prefixes(&context),
+                vec![
+                    "git@other.git.host:",
+                    "ssh://git@other.git.host/",
+                    "https://other.git.host/",
+                ]
+            );
+        }
+
+        #[test]
+        fn test_other_heroku_host() {
+            let mut context = default_context();
+            context.env.insert("HEROKU_HOST", "other.host".into());
+
+            assert_eq!(
+                git_prefixes(&context),
+                vec![
+                    "git@other.host:",
+                    "ssh://git@other.host/",
+                    "https://git.other.host/",
+                ]
+            );
+        }
+        #[test]
+        fn test_other_heroku_host_with_http() {
+            let mut context = default_context();
+            context
+                .env
+                .insert("HEROKU_HOST", "https://other.host".into());
+
+            assert_eq!(
+                git_prefixes(&context),
+                vec![
+                    "git@other.host:",
+                    "ssh://git@other.host/",
+                    "https://other.host/",
+                ]
+            );
+        }
+    }
+}
+
+/// find currently used Heroku app.
+/// Mimics:
+/// https://github.com/heroku/heroku-cli-command/blob/3e13ae899f64616c316dcf05398724fdd53c4d8a/src/flags/app.ts#L22-L40
+/// with
+/// https://github.com/heroku/heroku-cli-command/blob/5e13ae899f64616c316dcf05398724fdd53c4d8a/src/git.ts
+mod heroku_app {
+    use super::{heroku_vars, Context};
+    use git2::Repository;
+    use regex::Regex;
+
+    fn config_remote(repo: &Repository) -> Option<String> {
+        let config = repo.config().ok()?;
+        config.get_string("heroku.remote").ok()
+    }
+
+    fn get_apps_in_git_remotes(
+        context: &Context,
+        repo: &Repository,
+        only_remote: Option<String>,
+    ) -> Result<Vec<String>, git2::Error> {
+        let prefixes = heroku_vars::git_prefixes(context);
+        let remote_regexes: Vec<Regex> = prefixes
+            .iter()
+            .filter_map(|prefix| {
+                let regex = format!(r"{}(.*)\.git", prefix);
+
+                match Regex::new(&regex) {
+                    Ok(re) => Some(re),
+                    Err(err) => {
+                        log::debug!(
+                            "could not parse regex \"{}\" for Heroku remotes: {:?}",
+                            regex,
+                            err
+                        );
+                        None
+                    }
+                }
+            })
+            .collect();
+
+        let mut app_remotes = Vec::new();
+        for remote_name in repo.remotes()?.iter().flatten() {
```

Comment:
should the remotes be added to the `Context` and fetched with the other git details? 

Probably depends on how expensive opening the repo is